x-pack/plugins/fleet/public/components/agent_enrollment_flyout/agent_policy_select_create.tsx

```diff
@@ -16,43 +16,60 @@ import { AgentPolicyCreateInlineForm } from '../../applications/fleet/sections/a
 import type { AgentPolicy } from '../../types';
 import { incrementPolicyName } from '../../services';
 
-import { EnrollmentStepAgentPolicy } from '.';
+import { AgentPolicySelection } from '.';
 
 interface Props {
   agentPolicies: AgentPolicy[];
+  selectedPolicy?: AgentPolicy;
+  setSelectedPolicyId: (agentPolicyId?: string) => void;
   excludeFleetServer?: boolean;
-  onAgentPolicyChange: (key?: string, policy?: AgentPolicy) => void;
   withKeySelection: boolean;
   selectedApiKeyId?: string;
   onKeyChange?: (key?: string) => void;
   isFleetServerPolicy?: boolean;
-  policyId?: string;
+  refreshAgentPolicies: () => void;
 }
 
 export const SelectCreateAgentPolicy: React.FC<Props> = ({
   agentPolicies,
   excludeFleetServer,
-  onAgentPolicyChange,
+  setSelectedPolicyId,
```

Comment:
`setSelectedPolicyId` is not always set when using `<SelectCreateAgentPolicy>` , maybe that is causing the undefined error

Reply:
I think you are right there, I think I fixed it